### PR TITLE
Add register_table procedure support for iceberg using existing metadata

### PR DESCRIPTION
**Pull request by krvikash**
## Description

Adding `register_table` procedure for iceberg connector to register table using existing metadata.

- Iceberg table will be created using provided table location
- Look for the latest metadata file ending with `.metadata.json` inside the `metadata` folder and use it for creating the table
- If provided location is invalid/does not exist then throw an exception
- If no metadata file exists inside provided table location then throw an exception
- If more than one metadata file exists with the latest sequence number then throw an exception
- `schema_name`, `table_name`, and `table_location` should be not-null and valid, Otherwise, the exception will be thrown
-  User can optionally provide the valid `metadata_file_location` (See the usage below)

Valid usages:
 - `register_table(schema_name => ..., table_name => ..., table_location => ...)`
 - `register_table(schema_name => ..., table_name => ..., table_location => ..., metadata_file_location => ...)`

 Sample Queries:
`CALL iceberg.system.register_table('default', 'src_22', 'hdfs://hadoop-master:9000/user/hive/warehouse/orders_5-581fad8517934af6be1857a903559d44');`
 
`CALL iceberg.system.register_table('default', 'src_22', 'hdfs://hadoop-master:9000/user/hive/warehouse/orders_5-581fad8517934af6be1857a903559d44', null);`

 `CALL iceberg.system.register_table('default', 'src_22', 'hdfs://hadoop-master:9000/user/hive/warehouse/orders_5-581fad8517934af6be1857a903559d44', '00003-409702ba-4735-4645-8f14-09537cc0b2c8.metadata.json');`

  TODO:
  Add test cases

## Non-technical explanation

NA

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
(X) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findepi** [comment]
Added empty commit to retrigger the build after https://github.com/trinodb/trino/pull/14323 merged.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java [hunk 1]
```diff
@@ -623,4 +636,111 @@ private static void validateOrcBloomFilterColumns(ConnectorTableMetadata tableMe
...
+     * Validate provided location
+     *
+     * @param fileSystem TrinoFileSystem
+     * @param location location to validated
+     */
+    public static void validateLocation(TrinoFileSystem fileSystem, String location)
+    {
+        try {
+            if (!fileSystem.newInputFile(location).exists()) {
+                throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, format("Location %s does not exist", location));
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java:739]: The error code is not appropriate. The file  system is fine, is just that the file at the location provided is missing.

**Comment by findinpath** [comment]
Let's add some tests.

Note that `io.trino.plugin.iceberg.catalog.TrinoCatalog#dropTable` implementation do delete the table data (and metadata) as well.

The metadata files contain full path names for the `manifest-list` property.
One way to test this functionality is to create a table via SQL and **programatically** drop its content from the metastore so that we avoid to delete the content of the table . In this fashion, the table data remains intact and we can proceed to register it again and eventually make queries on the table.

You can find inspiration on how to get access within the integration tests to the metastore / aws client directly by looking at these tests:

- https://github.com/trinodb/trino/blob/5ac5fdb5f079f532b28128bf0bf9bf158fb8d55f/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreTableOperationsInsertFailure.java
- https://github.com/trinodb/trino/blob/7893908c1afe4da84faa245494f3ee4c6e458ccd/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueTableOperationsInsertFailure.java

**Comment by krvikash** [comment]
This PR got closed because of my mistake during running GitHub commands. Reopen a new PR https://github.com/trinodb/trino/pull/14325

@findinpath, Addressed your review comments in https://github.com/trinodb/trino/pull/14325

**Comment by alexjo2144** [comment]
> This PR got closed because of my mistake during running GitHub commands

For next time, you can reopen the same PR. There's a button at the bottom of the page

**Pull request closed without merging**